examples/tutorial 1 - Using a Quantum Device to Extract Machine-Learning Features.ipynb

```diff
@@ -0,0 +1,335 @@
+    "\n",
+    "compiler = qek_graphs.MoleculeGraphCompiler()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This library provides other compilers from other formats of graphs."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Creating and executing a feature extractor from an emulator\n",
+    "\n",
+    "The easiest way to process a graph is to compile and execute it for an emulator. QEK is built on top of Pulser, which provides several emulators. The simplest of these emulators is the `QutipEmulator`, which QEK uses for the `QutipExtractor`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import qek.data.extractors as qek_extractors\n",
+    "\n",
+    "# Use the Qutip Extractor.\n",
+    "extractor = qek_extractors.QutipExtractor(\n",
+    "    # Once computing is complete, data will be saved in this file.\n",
+    "    path=\"saved_data.json\",\n",
+    "    compiler=compiler\n",
+    ")\n",
+    "\n",
+    "# Add the graphs using the compiler we've picked previously.\n",
+    "extractor.add_graphs(graphs=og_ptcfm)\n",
+    "\n",
+    "# We may now compile them.\n",
+    "compiled = extractor.compile()\n",
+    "display(\"Compiled %s sequences\" % (len(compiled), ))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As you can see, the number of sequences compiled is lower than the number of samples loaded. Some of this is due to limitations within the algorithm (not all graphs can be efficiently laid out for execution on a Quantum Device), while others are due to the limitations of the emulator we target (which at the time of this writing is limited to 50 qubits).\n",
+    "\n",
+    "We may now run the extraction on the emulator:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Limit the number of qubits for this run, for performance reasons.\n",
+    "# You can increase this value to higher number of qubits, but this\n",
+    "# notebook will take longer to execute and may run out of memory.\n",
+    "MAX_QUBITS = 5\n",
+    "processed_dataset = await extractor.run(max_qubits=MAX_QUBITS) # Don't forget to `await`!\n",
+    "display(\"Extracted features from %s samples\"% (len(processed_dataset), ))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If you wish to extract features from more samples, feel free to increase the value of `MAX_QUBITS` above. However, you will soon run into limitations of a quantum emulator, and possibly crash this notebook. At this point, you have other options, such as using `EmuMPSExtractor` instead of `QutipExtractor`, a more recent emulator that features much better performance in most cases, or you can run the extraction on a physical QPU."
```

Comment:
Wow there is no graceful exit for this ?

Reply:
What do you mean?

Jupyter itself doesn't seem to break, but the underlying Python code raises weird error messages which I parse as "I've run out of memory".

Reply:
We probably should, yes.

Filed #42.